lib/iris/coords.py

```diff
@@ -1930,28 +1932,37 @@ def _new_points_requirements(self, points):
         Confirm that a new set of coord points adheres to the requirements for
         :class:`~iris.coords.DimCoord` points, being:
             * points are scalar or 1D,
-            * points are numeric, and
+            * points are numeric,
+            * points are not masked, and
             * points are monotonic.
 
         """
         if points.ndim not in (0, 1):
-            raise ValueError(
-                'The points array must be scalar or 1-dimensional.')
+            emsg = 'The {!r} {} points array must be scalar or 1-dimensional.'
+            raise ValueError(emsg.format(self.name(), self.__class__.__name__))
         if not np.issubdtype(points.dtype, np.number):
-            raise ValueError('The points array must be numeric.')
+            emsg = 'The {!r} {} points array must be numeric.'
+            raise ValueError(emsg.format(self.name(), self.__class__.__name__))
+        if ma.isMaskedArray(points) and ma.count_masked(points):
```

Comment:
Should probably be using 'is_masked' for this : https://docs.scipy.org/doc/numpy/reference/generated/numpy.ma.is_masked.html

Reply:
Yup, thanks @pp-mo ... I always forget that pattern, and it gives me the same behaviour that we want, namely:
```python
>>> import numpy as np
>>> import numpy.ma as ma
>>> a = np.arange(5)
>>> b = ma.arange(5)
>>> c = ma.arange(5)   
>>> c[0] = ma.masked
>>> ma.is_masked(a)
False
>>> ma.is_masked(b)
False
>>> ma.is_masked(c)
True
```

Reply:
@pp-mo Done 👍